src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java

```diff
@@ -166,4 +183,62 @@ private JsonArray listTemplates(String pageToken, JsonArray jsonElements) throws
     }
     return jsonElements;
   }
+
+  private List<StructuredRecord> listTemplateData(String templateName)
+    throws IOException, AribaException, InterruptedException {
+    AribaServices aribaServices = new AribaServices(config);
+    try {
+      accessToken = aribaServices.getAccessToken();
+    } catch (AribaException e) {
+      throw new IOException("unable to generate access token", e);
+    }
+    Schema schema = aribaServices.buildOutputSchema(accessToken, templateName);
+    URL dataURL = HttpUrl.parse(config.getBaseURL()).
+      newBuilder()
+      .addPathSegments(METADATA_PATH)
+      .addPathSegments(String.format(DATA_PATH_SEGMENT, config.getSystemType()))
+      .addPathSegments(templateName)
+      .addQueryParameter(ResourceConstants.PRODUCT, ResourceConstants.ANALYTICS)
+      .addQueryParameter(ResourceConstants.REALM, config.getRealm()).build().url();
+    Request request = new Request.Builder().get().url(dataURL).addHeader(ResourceConstants.API_KEY, config.getApiKey())
+      .addHeader(AUTHORIZATION, aribaServices.getAuthenticationKey(accessToken)).build();
+    OkHttpClient enhancedOkHttpClient = new OkHttpClient();
+    Response response = enhancedOkHttpClient.newCall(request).execute();
+    AribaResponseContainer aribaResponseContainer = aribaServices.tokenResponse(response);
+    StructuredRecord structuredRecord = null;
```

Comment:
The object structuredRecord is not required in this scope. Instead we can directly use 'recordList.add(transformer.readFields(record, schema))' later.

Reply:
Done